opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1516,610 +2031,1764 @@ void CpGrid::addLgrsUpdateLeafView(const std::vector<std::array<int,3>>& cells_p
+            const auto& elemLevelIdx = (elemLevel == 0) ? element.getOrigin().index() : element.getLevelElem().index(); // parent cell index in the level it was born.
+            adapted_child_to_parent_cells[cell] = {elemLevel, elemLevelIdx};
+            adapted_cell_to_idxInParentCell[cell] = adaptedCell_to_elemLgrAndElemLgrCell[cell][1];
+        }
+        else {
+            // Populate child_to_parent_ relationships from the previous (existing grids before calling adapt).
+            assert(elemLgr == -1);
+            const auto& elemIdx = adaptedCell_to_elemLgrAndElemLgrCell[cell][1];
+            // Search for the level where the parent cell was born, and its index in that level grid.
+            // Notice that elemLgr is a cell index in current_view_data_ (the starting grid where elements got marked for (further) refinement).
+            const auto& element = Dune::cpgrid::Entity<0>(*current_view_data_, elemIdx, true); // elemIdx of equivalent cell in the starting grid.
+            if(element.hasFather()) {
+                adapted_child_to_parent_cells[cell] = {element.father().level(), element.father().index()};
+                const auto& elemLevel = element.level(); // parent cell level (level where the parent cell was born, might not coincide with the starting grid).
+                const auto& elemLevelIdx = (elemLevel == 0) ? element.getOrigin().index() : element.getLevelElem().index(); // parent cell index in the level it was born.
+                adapted_cell_to_idxInParentCell[cell] = this-> data_[elemLevel]-> cell_to_idxInParentCell_[elemLevelIdx];
+            }
+        }
+    }
+
+    for (int shiftedLevel = 0; shiftedLevel < static_cast<int>(refined_cell_count_vec.size()); ++shiftedLevel) {
+        refined_child_to_parent_cells_vec[shiftedLevel].resize(refined_cell_count_vec[shiftedLevel]);
+        refined_cell_to_idxInParentCell_vec[shiftedLevel].resize(refined_cell_count_vec[shiftedLevel]);
+        const auto& level =  shiftedLevel + startingGridIdx +1;
+        // Every refined cell has a parent cell
+        for (int cell = 0; cell < refined_cell_count_vec[shiftedLevel]; ++cell) {
+            const auto& elemLgr = refinedLevelAndRefinedCell_to_elemLgrAndElemLgrCell[{level, cell}][0];
+            // (elemLgr == -1) means that the adapted cell is equivalent to a cell from the starting grid (current_view_data_)
+            assert(elemLgr != -1);
+            // Search for the level where the parent cell was born, and its index in that level grid.
+            // Notice that elemLgr is a cell index in current_view_data_ (the starting grid where elements got marked for (further) refinement).
+            const auto& element = Dune::cpgrid::Entity<0>(*current_view_data_, elemLgr, true);  // elemLgr == parent cell index in starting grid.
+            const auto& elemLevel = element.level(); // parent cell level (level where the parent cell was born, might not coincide with the starting grid).
```

Comment:
In the current version, the code it's a bit simpler. In any case, elemLgr represent the element index of a cell that got marked to be refined, in the "starting grid", current_view_data_. On the other hand, elemLevel here represents the level where the marked element with index "elemLgr" was born. This might belevel 0, or any preAdapt refined-level-grid. If current_view_data_ coincides with level zero grid, then elemLevel is equal to 0. If current_view_data_ represents a mixed grid with coarse and refined cells, with "grid.maxLevel() == 3", then elemLevel could be 0,1,2, or 3, depending on where it was born.